modules/audit-enqueued-assets/load.php

```diff
@@ -0,0 +1,312 @@
+ */
+function perflab_aea_get_total_enqueued_scripts() {
+	$enqueued_scripts      = false;
+	$list_enqueued_scripts = get_transient( 'aea_enqueued_scripts' );
+	if ( $list_enqueued_scripts ) {
+		$enqueued_scripts = count( $list_enqueued_scripts );
+	}
+	return $enqueued_scripts;
+}
+
+/**
+ * Gets total size in bytes of Enqueued Scripts.
+ *
+ * @since 1.0.0
+ *
+ * @return int Byte Total size.
+ */
+function perflab_aea_get_total_size_bytes_enqueued_scripts() {
+	$total_size            = 0;
+	$list_enqueued_scripts = get_transient( 'aea_enqueued_scripts' );
+	if ( $list_enqueued_scripts ) {
+		foreach ( $list_enqueued_scripts as $enqueued_script ) {
+			$total_size += $enqueued_script['size'];
+		}
+	}
+	return $total_size;
+}
+
+/**
+ * Gets total of enqueued styles.
+ *
+ * @since 1.0.0
+ *
+ * @return int|false Number of total styles or false if transient hasn't been set.
+ */
+function perflab_aea_get_total_enqueued_styles() {
+	$enqueued_styles      = false;
+	$list_enqueued_styles = get_transient( 'aea_enqueued_styles' );
+	if ( $list_enqueued_styles ) {
+		$enqueued_styles = count( $list_enqueued_styles );
+	}
+	return $enqueued_styles;
+}
+
+/**
+ * Gets total size in bytes of Enqueued Styles.
+ *
+ * @since 1.0.0
+ *
+ * @return int Byte Total size.
+ */
+function perflab_aea_get_total_size_bytes_enqueued_styles() {
+	$total_size           = 0;
+	$list_enqueued_styles = get_transient( 'aea_enqueued_styles' );
+	if ( $list_enqueued_styles ) {
+		foreach ( $list_enqueued_styles as $enqueued_style ) {
+			$total_size += $enqueued_style['size'];
+		}
+	}
+	return $total_size;
+}
+
+/**
+ * Adds tests to site health.
+ *
+ * @since 1.0.0
+ *
+ * @param array $tests Site Health Tests.
+ * @return array
+ */
+function perflab_aea_add_enqueued_assets_test( $tests ) {
+	$tests['direct']['enqueued_js_assets']  = array(
+		'label' => esc_html__( 'JS assets', 'performance-lab' ),
+		'test'  => 'perflab_aea_enqueued_js_assets_test',
+	);
+	$tests['direct']['enqueued_css_assets'] = array(
+		'label' => esc_html__( 'CSS assets', 'performance-lab' ),
+		'test'  => 'perflab_aea_enqueued_css_assets_test',
+	);
+
+	return $tests;
+}
+add_filter( 'site_status_tests', 'perflab_aea_add_enqueued_assets_test' );
+
+/**
+ * Callback for enqueued_js_assets test.
+ *
+ * @since 1.0.0
+ *
+ * @return array
+ */
+function perflab_aea_enqueued_js_assets_test() {
+	/**
+	 * If the test didn't run yet, deactivate.
+	 */
+	$enqueued_scripts = perflab_aea_get_total_enqueued_scripts();
+	if ( false === $enqueued_scripts ) {
+		return array();
+	}
+
+	$result = array(
+		'label'       => esc_html__( 'Enqueued JS assets', 'performance-lab' ),
+		'status'      => 'good',
+		'badge'       => array(
+			'label' => esc_html__( 'Performance', 'performance-lab' ),
+			'color' => 'blue',
+		),
+		'description' => sprintf(
+			'<p>%s</p>',
+			esc_html__( 'The amount of enqueued JS assets is acceptable.', 'performance-lab' )
```

Comment:
I think it would be worth still informing about the exact numbers here as well, i.e. providing the number of scripts and their overall file size here. Maybe the sentence could be like this: `The amount of %1$s enqueued scripts (size: %2$s) is acceptable.`

Nit-pick, but we should also use a consistent term throughout all messaging, i.e. "scripts" instead of "JS assets".

Reply:
implemented